src/web/mjs/connectors/Sukima.mjs

```diff
@@ -0,0 +1,191 @@
+                        method: 'POST',
+                        body: '{"free":["1","2"],"page":'+page+',"sort_by":"0","tag":["'+uri.searchParams.get('tag')+'"]}',
+                        headers: {
+                            'Content-Type': 'application/json;charset=utf-8',
+                        }
+                    });
+
+                    let pageContent = await this.fetchJSON(request);
+                    pages = pageContent.max_page;
+
+                    for (const manga of pageContent.items) {
+                        mangas.push(
+                            {
+                                id: '/api/book/v1/title/'+manga.title_code+'/',
+                                title: manga.title_name
+                            }
+                        );
+                    }
+                }
+            }
+        }
+
+        return mangas;
+    }
+
+    async _getChapters(manga) {
+        let request = new Request(new URL(manga.id, this.url), {
+            method: 'POST',
+            body: '{}',
+            headers: {
+                'Content-Type': 'application/json;charset=utf-8',
+            }
+        });
+
+        let chapters = [];
+        let books = await this.fetchJSON(request);
+        for (const book of books.contents) {
+            for (const chapter of book.stories) {
+                chapters.push(
+                    {
+                        id: '/bv/t/' + chapter.title_code + '/v/' + chapter.volume + '/s/' + chapter.story + '/p/0',
+                        title: chapter.volume.toString().padStart(3, '0') + '-' + chapter.story.toString().padStart(3, '0') + ' - ' + chapter.info.text.trim()
+                    }
+                );
+            }
+        }
+        return chapters;
+    }
+
+    async _getPages(manga) {
+        let script = `
+            new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    let shuffle_data = [];
+                    for (page = 0; page < PAGES_INFO.length; page++) {
+                        shuffle_data.push(
+                            {
+                                id: PAGES_INFO[page].page_url,
+                                blocklen: BLOCKLEN,
+                                shuffle_map: JSON.parse(PAGES_INFO[page].shuffle_map)
+                            }
+                        )
+                    }
+                    resolve(shuffle_data);
+                }, 2500);
+            });
+        `;
+        let request = new Request(new URL(manga.id, this.url));
+        let data = await Engine.Request.fetchUI(request, script);
+
+        return data.map(page => {
+            return this.createConnectorURI(
+                {
+                    id: page.id,
+                    blocklen: page.blocklen,
+                    shuffle_map: page.shuffle_map,
+                }
+            );
+        });
+    }
+
+    async _handleConnectorURI(payload) {
+        while (this.lock) {
```

Comment:
Is this lock really required, this is a dangerous bottleneck.
A user may click the preview button for a chapter with e.g. 20 images and the method is now locked for at least 20 seconds... After seeing the cover, the use decides to look at the cover of the next chapter and clicks the preview... The method will be locked for another amount of time.

A better approach would be to check the response status and repeat the method call after a small amount of time if the request failed.
See: https://github.com/manga-download/hakuneko/blob/40fe23c9bcd71bd453a6ebf9a331250e9a9f02c0/src/web/mjs/connectors/templates/MangaEden.mjs#L87

Reply:
My problem while testing was that I got blocked for several hours after fetching all images 'at once'. And since `this.config` throttle is not applicable I introduced a mechanism to only process one image at a time. I also double checked and the page is indeed only loading the next image when you flip a page. Therefore I added a reasonable delay between pages. It maybe a shortcoming of my implementation but I think it mimics the website usage.

Reply:
Could it be that the reason for the temp ban was the amount of the requested images, were missing the referer header? Maybe there is no ban when adding the correct referer for each image (the HTML page of the chapter)
![image](https://user-images.githubusercontent.com/27136773/83322854-2f72d580-a25b-11ea-9ed2-93c7a40b90c6.png)

Reply:
This seems to do the trick. Nice catch!